v4/export/dump.go

```diff
@@ -36,6 +36,13 @@ func Dump(conf *Config) (err error) {
 	if err != nil {
 		return err
 	}
+	if conf.ServerInfo.ServerType == ServerTypeTiDB && conf.ServerInfo.ServerVersion.Compare(*tidbV312) >= 0 {
+		log.Info("set tidb_mem_quota_query", zap.Uint64("tidb_mem_quota_query", conf.TiDBMemQuotaQuery))
+		_, err = pool.Exec("set tidb_mem_quota_query = ?", conf.TiDBMemQuotaQuery)
```

Comment:
should this be moved into `getDSN`?

Reply:
sure

Reply:
wait a minute, we must make connected db is tidb and then we can set this session variable. how to do this?

Reply:
🤔 In Lightning if we get the error we'll delete the variable and reconnect. You could use https://pkg.go.dev/github.com/go-sql-driver/mysql?tab=doc#Config.FormatDSN to construct the DSN.

Alternatively, execute this statement every time we perform a `SELECT *`? 

I suspect https://github.com/pingcap/dumpling/issues/104#issuecomment-646445174 is because this `SET` is run on only one connection rather than every connection.

Reply:
This is caused because `_tidb_rowid` is distributed sparsely.
![image](https://user-images.githubusercontent.com/21104942/85105649-5fe7d700-b23d-11ea-9ff1-a5f553583d68.png)

Reply:
I don't think that's relevant to the OOM

Reply:
I tried and it will not return error

Reply:
> I don't think that's relevant to the OOM

Why? TiDB gets oom because it can't handle a wide ranged select sql